### ISSUE TITLE
Navigation distance unit should default to route option's `distanceMeasurementSystem`

### ISSUE THREAD

**Issue by alessandrodn** [mapbox/mapbox-navigation-ios#2578]
**Mapbox Navigation SDK version:** 0.40

# Introduction
I want to have English as language and Metric as the measurement unit: to achieve that I set route options locale to `en_US` and `distanceMeasurementSystem` to `.metric`.

Without any other modification, I get the navigation instruction in English/Metric while the UI alerts are still in Imperial.

I expect the route options to be the single source of truth (as described in #1055).

# Fix
I would expect that the `NavigationViewController` uses the route options `distanceMeasurementSystem` to set the NavigationService's distanceUnit, while it still read use locale: from `NavigationViewController.swift`

```swift
NavigationSettings.shared.distanceUnit = routeOptions.locale.usesMetric ? .kilometer : .mile
```

Shouldn't this line be something like this, or am I doing something wrong?

```swift
NavigationSettings.shared.distanceUnit = routeOptions. distanceMeasurementSystem == .metric ? .kilometer : .mile
```

I'm willing to open a PR in this direction if agreed.

# Workaround
As a workaround, I explicitly override the `NavigationSettings`'s `distanceUnit` after the creation of the NavigationViewController.

**Comment by S2Ler** [comment]
@alessandrodn Sorry, this issue was lost in a wind of changes. Thanks for reporting! 

I definitely see the inconsistency here and we'll prioritize this issue.

**Comment by S2Ler** [comment]
@alessandrodn The fix will be part of the v2.1 release. You can try the v2.0.0-rc.1, which is scheduled for release today. 

Thanks for the contribution!
